content/opt/setup/30-locales

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+if [[ ! -f /etc/locale.gen.dist ]]
+then
+   cp /etc/locale.gen /etc/locale.gen.dist
+fi
+
+echo "BUILD_LOCALES: $BUILD_LOCALES"
+typeset -i m=$(echo $BUILD_LOCALES | awk -F ',' '{print NF}')
+typeset -i n=1
+while [[ $n -le $m ]]
```

Comment:
You're iterating here over an index. Well, that's possible, but in bash this is discouraged. The language is not made for this. Use things like an enhanced for loop.

```
for elem in $(command_which spits out elements); do
   # do some stuff with "${elem}"
done
```

Rule of thumb: At the point where you have to use real integers either

- bash isn't the way to go and another language fits much better
- you misuse bash
- you know that you misuse bash, but it's the only language available